Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFrame.cpp

```diff
@@ -63,10 +64,27 @@ static void webkit_frame_class_init(WebKitFrameClass*)
 {
 }
 
+static CString getURL(WebFrame* webFrame)
+{
+    auto* documentLoader = webFrame->coreFrame()->loader().provisionalDocumentLoader();
+    if (!documentLoader)
+        documentLoader = webFrame->coreFrame()->loader().documentLoader();
+
+    ASSERT(documentLoader);
+
+    if (!documentLoader->unreachableURL().isEmpty())
+        return documentLoader->unreachableURL().string().utf8();
+
+    return documentLoader->url().string().utf8();
+}
+
 WebKitFrame* webkitFrameCreate(WebFrame* webFrame)
 {
     WebKitFrame* frame = WEBKIT_FRAME(g_object_new(WEBKIT_TYPE_FRAME, NULL));
     frame->priv->webFrame = webFrame;
+
+    webkitFrameSetURI(frame, getURL(webFrame));
```

Comment:
No need to compare uris here, we can do `frame->priv->uri = getURL(webFrame);` directly.